sym_object.py

```diff
@@ -567,3 +577,38 @@ def updateName(self, newName):
 
         # update member variable
         self.name = newName
+
+
+    def addSubObject(self, child):
+        '''add child to parent's (self's) UI object and setting parameters'''
+        child.resizeUIObject(self, 1, child.width)
+        child.parent_name = self.name
+        child.z = self.z + 1
+        if not child.name in self.connected_objects:
+            self.connected_objects.append(child.name)
+
+        for object in self.state.sym_objects.values():
+            object.setZValue(object.z)
+
+    def resizeParent(self, child):
+        '''reduce the size of the parent if it has one child'''
+        self.removeFromGroup(self.rect)
```

Comment:
these happen a couple of times in UI resizing, intitialization, etc. i know this is probably a derived QT function but if the items being modified are the same, can we move these batch updates to its own function? could be wrong

Reply:
Sorry, in this case im referring to the successive removeFromGroup calls

Reply:
Good point, will refactor those into its own function.